src/panoramic/cli/cli.py

```diff
@@ -211,3 +211,112 @@ def scaffold(target_dataset: str, yes: bool):
+@click.option('--user', '-u', type=str, help='Connection username.')
+@click.option('--password', '-p', type=str, help='Connection password.')
+@click.option('--password-stdin', is_flag=True, help='Read connection password from standard input.')
+@click.option('--host', '-H', type=str, help='Connection hostname.')
+@click.option('--port', '-p', type=str, help='Connection port.')
+@click.option('--database-name', '-d', type=str, help='Connection database name.')
+@click.option('--no-test', '-n', is_flag=True, help='Do NOT try test the connection.')
+def create(name, type, user, host, port, password, password_stdin, database_name, no_test):
+    """Add new data connection.
+
+    pano data-connections create postgres-prod --type postgres --user my_user \\
+     --password-stdin --host localhost --port 5432 --database-name my_db
+    """
+    from panoramic.cli.data_connections import create_data_connection_command
+
+    create_data_connection_command(name, type, user, host, port, password, password_stdin, database_name, no_test)
+
+
+@data_connections.command()
+@click.argument('name', type=str)
+@click.option('--type', '-t', type=str, help='Type of the database. E.g. "postgres", "snowflake".')
+@click.option('--user', '-u', type=str, help='Connection username.')
+@click.option('--password', '-p', type=str, help='Connection password.')
+@click.option('--password-stdin', is_flag=True, help='Read connection password from standard input.')
+@click.option('--host', '-h', type=str, help='Connection hostname.')
+@click.option('--port', '-p', type=str, help='Connection port.')
+@click.option('--database-name', '-d', type=str, help='Connection database name.')
+@click.option('--no-test', '-n', is_flag=True, help='Do NOT try test the connection.')
+def update(name, type, user, host, port, password, password_stdin, database_name, no_test):
+    """Update existing data connection.
+
+    pano data-connections update postgres-prod --database-name my_new_prod_db
+    """
+    from panoramic.cli.data_connections import update_data_connection_command
+
+    update_data_connection_command(name, type, user, host, port, password, password_stdin, database_name, no_test)
+
+
+@data_connections.command()
+@click.argument('name', type=str)
+def remove(name):
+    """Remove existing data connection.
+
+    pano data-connections remove postgres-prod
+    """
+    from panoramic.cli.data_connections import remove_data_connection_command
+
+    remove_data_connection_command(name)
+
+
+@data_connections.command(name='list')
+@click.option('--show-password', is_flag=True, help='Show passwords.')
+def list_(show_password):  # we cannot have method 'list' due to conflicts
+    """List all available data connections.
+
+    pano data-connections list --show-password
+    """
+    from panoramic.cli.data_connections import list_data_connections_command
+
+    list_data_connections_command(show_password)
+
+
+@data_connections.command()
+def test():
```

Comment:
Nice 👏 . We could add an optional `name` arg as a followup.

Reply:
added optional name arg